R/apa_print_merMod.R

```diff
@@ -1,144 +1,112 @@
-#'   }
-#'   if(in_paren) res$statistic <- in_paren(res$statistic)
-#'
-#'   res$full_result <- as.list(paste0(res$estimate, ", ", res$statistic))
-#'
-#'   names(res$full_result) <- names(res$statistic) <- names(res$estimate) <- sanitize_terms(res_table$term)
-#'
-#'   # return
-#'   res
-#' }
-#'
-#' #' @rdname apa_print.merMod
-#' #' @method apa_print mixed
-#' #' @export
-#'
-#' apa_print.mixed <- function(x, ...) {
-#'
-#'   anova_table <- x$anova_table
-#'   attr(anova_table, "method") <- attr(x, "method")
-#'   apa_print(anova_table, ...)
-#' }
+#' Format Statistics from Hierarchical (Generalized) Linear Models (APA 6th edition)
+#'
+#' These methods take objects from various R functions that calculate
+#' hierarchical (generalized) linear models to create formatted character strings
+#' to report the results in accordance with APA manuscript guidelines.
+#'
+#' @param x A fitted hierarchical (generalized) linear model, either from [lme4::lmer()],
+#'   [lmerTest::lmer()], [afex::mixed()], or [lme4::glmer()].
+#' @param args_confint Named list. Additional arguments that are passed to [lme4::confint.merMod()], see details.
+#' @param est_name     An optional character. The label to be used for fixed-effects coefficients.
+#' @inheritParams print_anova
+#' @param ... Further arguments that are passed to [printnum()].
+#' @details
+#'   Confidence intervals are calculated by calling [lme4::confint.merMod()]. By
+#'   default, *profile* confidence intervals are calculated, except for generalized
+#'   linear models with non-fixed scale. In such cases, *bootstrap* confidence intervals
+#'   are calculated by default. See [lme4::confint.merMod()] for details.
+#'
+#' @family apa_print
+#' @rdname apa_print.merMod
+#' @method apa_print merMod
+#' @export
+#' @md
+
+apa_print.merMod <- function(
+  x
+  , args_confint
+  , in_paren = FALSE
+  , est_name
+  , ...
+) {
+
+  # Input validation and processing ----
+  args <- list(...)
+
+  if(missing(args_confint) || is.null(args_confint)) args_confint <- list()
+  validate(args_confint, check_class = "list")
+
+  # `in_paren` is validated in `glue_apa_results()`
+
+  if(missing(est_name) || is.null(est_name)) {
+    est_name <- "$\\hat{\\beta}$"
+  } else {
+    validate(est_name, check_class = "character", check_length = 1L)
+    est_name <- paste0("$", strip_math_tags(est_name), "$")
+  }
+
+  # GLMM with non-fixed scale? (cf. lme4::profile.merMod)
+  no_profile <- lme4::isGLMM(x) && x@devcomp$dims[["useSc"]]
+
+  args_confint <- defaults(
+    args_confint
+    , set = list(
+      object = x
+      , parm = "beta_"
+    )
+    , set.if.null = list(
+      level = .95
+      , method = if(no_profile){"boot"} else {"profile"}
+      , nsim = 2e3L
+    )
+  )
+
+
+
+
+  # Rearrange ----
+  x_summary <- summary(x)
+
+  res_table <- as.data.frame(
+    x_summary$coefficients
+    , row.names = NULL
+  )
+
+  res_table$Term <- rownames(x_summary$coefficients)
```

Comment:
Don't we use `term` instead of `Term`?

Reply:
I do this because the lookup tables contain `Term = "term"`for column names  and `Term = "Term" for labels (similar to `Effect = "term"). So basically, here I mimic the input that `canonize()` is typically confronted with (instead of creating new cases in the lookup tables). The final result is a column with name `term` and label `Term`